aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/kinesisvideo/http/HostnameVerifyingX509ExtendedTrustManager.java

```diff
@@ -0,0 +1,233 @@
+package com.amazonaws.kinesisvideo.http;
+
+import com.amazonaws.kinesisvideo.common.logging.Log;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509ExtendedTrustManager;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+/*
+     Adapted from (Apache 2.0 License):
+
+     https://github.com/apache/zookeeper/blob/master/zookeeper-server/src/main/java/org/apache/zookeeper/common/
+     ZKTrustManager.java
+
+     A custom TrustManager that supports hostname verification via org.apache.http.conn.ssl.DefaultHostnameVerifier.
+     *
+     * We attempt to perform verification using just the IP address first and if that fails will attempt to perform a
+     * reverse DNS lookup and verify using the hostname.
+*/
+
+public class HostnameVerifyingX509ExtendedTrustManager extends X509ExtendedTrustManager {
+
+    private static final HostnameVerifier DEFAULT_HOSTNAME_VERIFIER = HostnameVerifier.INSTANCE;
+    private Log log = new Log(Log.SYSTEM_OUT);
+    private final boolean clientSideHostnameVerificationEnabled;
+
+    private final X509ExtendedTrustManager x509ExtendedTrustManager;
+
+    public HostnameVerifyingX509ExtendedTrustManager(final boolean clientSideHostnameVerificationEnabled) {
+        this.x509ExtendedTrustManager = getX509ExtendedTrustManager();
+        this.clientSideHostnameVerificationEnabled = clientSideHostnameVerificationEnabled;
+    }
+
+    private X509ExtendedTrustManager getX509ExtendedTrustManager() {
+        TrustManagerFactory factory;
+        try {
+            factory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            factory.init((KeyStore) null);
+        } catch (NoSuchAlgorithmException nsae) {
+            log.error("Unable to initialize default TrustManagerFactory", nsae);
+            throw new RuntimeException(nsae);
```

Comment:
I'd suggest including a message as well, in addition to the Throwable.  Probably the same message you are logging as an error would work.  This suggestion applies to all 3 places where you are throwing a `RuntimeException`